smartsim/_core/schemas/utils.py

```diff
@@ -26,30 +50,30 @@ def _register(cls: t.Type[_SchemaT]) -> t.Type[_SchemaT]:
 
         return _register
 
-    def schema_to_dict(self, schema: _SchemaT) -> t.Dict[str, t.Any]:
+    def to_string(self, schema: _SchemaT) -> str:
+        return str(self._to_message(schema))
+
+    def _to_message(self, schema: _SchemaT) -> _Message[_SchemaT]:
         reverse_map = dict((v, k) for k, v in self._map.items())
         try:
             val = reverse_map[type(schema)]
         except KeyError:
             raise TypeError(f"Unregistered schema type: {type(schema)}") from None
-        # TODO: This method is deprectated in pydantic >= 2
-        dict_ = schema.dict()
-        dict_[self._type_name_key] = val
-        return dict_
+        return _Message(val, schema, self._msg_delim)
 
-    def serialize_to_json(self, schema: _SchemaT) -> str:
-        return json.dumps(self.schema_to_dict(schema))
-
-    def mapping_to_schema(self, obj: t.Mapping[t.Any, t.Any]) -> _SchemaT:
+    def from_string(self, str_: str) -> _SchemaT:
         try:
-            type_ = obj[self._type_name_key]
-        except KeyError:
-            raise ValueError(f"Could not parse object: {obj}") from None
+            type_, _ = str_.split(self._msg_delim, 1)
+        except ValueError:
+            _msg = f"Failed determine schema type of string {repr(str_)}"
```

Comment:
As I'm not the best candidate to judge English grammar, I deferred to ChatGPT, which says

> The sentence "Failed determine schema type of string" is not grammatically correct. It's missing a verb to connect "Failed" with "determine." To make the sentence grammatically correct, you could use one of the following options:
> - "Failed to determine the schema type of the string."
> - "Failed in determining the schema type of the string."
> 
> Adding "to" before "determine" and including "the" before "schema" and "string" makes the sentence more precise and clear.

Reply:
🤦 At this rate, no one is going to believe the English is my native language. Totally correct, will change!